### PR TITLE
Improved CLI for PVClaims

### PR DESCRIPTION
**Pull request by markturansky**
Resolves https://github.com/GoogleCloudPlatform/kubernetes/issues/12439

@smarterclayton

### CODE REVIEW

**Comment by markturansky** [comment]
Can someone give me insight into the Shippable build issue?

I've run hack/test-* and all pass for me.

**Comment by smarterclayton** [comment]
@kargakis can you take a look at this in case I don't get to it?

On Aug 10, 2015, at 11:56 PM, Mark Turansky <notifications@github.com>
wrote:

Resolves #12439
<https://github.com/GoogleCloudPlatform/kubernetes/issues/12439>

@smarterclayton <https://github.com/smarterclayton>
------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/GoogleCloudPlatform/kubernetes/pull/12512
Commit Summary

   - Improved CLI for PVClaims

File Changes

   - *M* pkg/kubectl/describe.go
   <https://github.com/GoogleCloudPlatform/kubernetes/pull/12512/files#diff-0>
   (20)
   - *M* pkg/kubectl/resource_printer.go
   <https://github.com/GoogleCloudPlatform/kubernetes/pull/12512/files#diff-1>
   (17)

Patch Links:

   - https://github.com/GoogleCloudPlatform/kubernetes/pull/12512.patch
   - https://github.com/GoogleCloudPlatform/kubernetes/pull/12512.diff

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/12512>.

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -815,13 +815,18 @@ func printPersistentVolumeClaim(pvc *api.PersistentVolumeClaim, w io.Writer, wit
...
-		}
+	labels := formatLabels(pvc.Labels)
+	phase := pvc.Status.Phase
+	storage := pvc.Spec.Resources.Requests[api.ResourceStorage]
+	capacity := int64(0)
+	accessModes := ""
+	if pvc.Spec.VolumeName != "" {
+		accessModes = volume.GetAccessModesAsString(pvc.Status.AccessModes)
+		storage = pvc.Status.Capacity[api.ResourceStorage]
+		capacity = storage.Value()
```
Comment (smarterclayton) [inline pkg/kubectl/resource_printer.go]: Do we want this to be using the units (ie .String())? Or does value() return 2Gi

Comment (markturansky) [inline pkg/kubectl/resource_printer.go]: I missed the pretty-print string version somehow and went for the int value.  

I added the string here and also to the PV resource_printer and describer.

Comment (smarterclayton) [inline pkg/kubectl/resource_printer.go]: Thanks, @kargakis for the remainder of review

On Aug 11, 2015, at 4:39 PM, Mark Turansky <notifications@github.com> wrote:

In pkg/kubectl/resource_printer.go
<https://github.com/GoogleCloudPlatform/kubernetes/pull/12512#discussion_r36795653>
:

> @@ -815,13 +815,18 @@ func printPersistentVolumeClaim(pvc *api.PersistentVolumeClaim, w io.Writer, wit
>  		}
>  	}
>
> -	if withNamespace {
> -		if _, err := fmt.Fprintf(w, "%s\t", namespace); err != nil {
> -			return err
> -		}
> +	labels := formatLabels(pvc.Labels)
> +	phase := pvc.Status.Phase
> +	storage := pvc.Spec.Resources.Requests[api.ResourceStorage]
> +	capacity := int64(0)
> +	accessModes := ""
> +	if pvc.Spec.VolumeName != "" {
> +		accessModes = volume.GetAccessModesAsString(pvc.Status.AccessModes)
> +		storage = pvc.Status.Capacity[api.ResourceStorage]
> +		capacity = storage.Value()

I missed the pretty-print string version somehow and went for the int
value.

I added the string here and also to the PV resource_printer and describer.

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/12512/files#r36795653>
.

**Comment by kargakis** [comment]
LGTM, needs rebase

cc: @mnagy

**Comment by markturansky** [comment]
Rebased and squashed.

**Comment by smarterclayton** [comment]
LGTM

**Pull request merged**